bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/internal/ThingManager.java

```diff
@@ -508,29 +509,130 @@ public Void call() throws Exception {
                     return null;
                 }
             });
-        } catch (ExecutionException ex) {
-            String message = ex.getCause() != null ? ex.getCause().getMessage() : ex.getMessage();
-            ThingStatusInfo statusInfo = buildStatusInfo(ThingStatus.UNINITIALIZED,
-                    ThingStatusDetail.HANDLER_INITIALIZING_ERROR, message);
-            setThingStatus(thing, statusInfo);
-            logger.error(
-                    "Exception occured while calling thing handler factory '" + thingHandlerFactory + "': " + message,
-                    ex.getCause());
         } catch (TimeoutException ex) {
             ThingStatusInfo statusInfo = buildStatusInfo(ThingStatus.UNINITIALIZED,
-                    ThingStatusDetail.HANDLER_INITIALIZING_ERROR, ex.getMessage());
+                    ThingStatusDetail.HANDLER_REGISTERING_ERROR,
+                    ex.getCause() != null ? ex.getCause().getMessage() : ex.getMessage());
             setThingStatus(thing, statusInfo);
             logger.warn("Registering handler for thing '{}' takes more than {}ms.", thing.getUID(),
                     SafeMethodCaller.DEFAULT_TIMEOUT);
         } catch (Exception ex) {
             ThingStatusInfo statusInfo = buildStatusInfo(ThingStatus.UNINITIALIZED,
-                    ThingStatusDetail.HANDLER_INITIALIZING_ERROR, ex.getMessage());
+                    ThingStatusDetail.HANDLER_REGISTERING_ERROR,
+                    ex.getCause() != null ? ex.getCause().getMessage() : ex.getMessage());
             setThingStatus(thing, statusInfo);
             logger.error("Exception occured while calling thing handler factory '" + thingHandlerFactory + "': "
                     + ex.getMessage(), ex);
         }
     }
 
+    private void initializeHandler(Thing thing) {
+        if (isInitializable(thing)) {
+            ThingStatusInfo statusInfo = buildStatusInfo(ThingStatus.INITIALIZING, ThingStatusDetail.NONE);
+            setThingStatus(thing, statusInfo);
+            initializeHandler(thing, thing.getHandler());
+        } else {
+            logger.debug("Thing '{}' not initializeable, check required configuration parameters.", thing.getUID());
+            ThingStatusInfo statusInfo = buildStatusInfo(ThingStatus.UNINITIALIZED,
+                    ThingStatusDetail.HANDLER_CONFIGURATION_PENDING);
+            setThingStatus(thing, statusInfo);
+        }
+    }
+
+    private boolean isInitializable(Thing thing) {
+        // determines if all 'required' configuration parameters are available in the configuration
+        logger.debug("Determine if thing '{}' is initializable.", thing.getUID());
```

Comment:
this is more a trace output - and you are not logging the result of this check. So either remove this, or set it to trace and add another trace message about the result before returning